### DISCUSSION TITLE
Performance improvement due to xml.Unmarshaller implementation

### DISCUSSION THREAD

**Discussion by xakep666** [TheDataShed/xlsxreader#33] [Ideas]
When I profiled this library I found that parser waste significant amount of time in `reflect` functions.
I found some interesting recipies in this post https://stackoverflow.com/a/61858457 and implemented parts 2 and 3 for shared strings and rows types.
In simple benchmark 
```go
func BenchmarkOpenReadFile(b *testing.B) {
	bytes, err := ioutil.ReadFile("big.xlsx")
	require.NoError(b, err)

	b.ReportAllocs()
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		f, err := NewReader(bytes)
		require.NoError(b, err)
		for range f.ReadRows(f.Sheets[0]) {}
	}
}
```
I've got approx 20-25% better time.
But it brings boilerplate code (error handling, token matching, etc.). Is this result good enough to make pull request and discuss about proposed changes?

**Comment by dglsparsons** [comment]
No harm in opening the PR - it sounds like a very good speed up, and I'd be very curious personally to see how much more verbose the changes make things.

**Reply by xakep666** [reply]
https://github.com/TheDataShed/xlsxreader/pull/34

**Reply by dglsparsons** [reply]
Thank you :)
